AdyenTwint/TwintComponent.swift

```diff
@@ -39,19 +126,49 @@ public final class TwintComponent: PaymentComponent {
         self.paymentMethod = paymentMethod
         self.context = context
         self.configuration = configuration
-
-        let details = TwintDetails(type: paymentMethod,
-                                   subType: "sdk")
-        self.paymentData = PaymentComponentData(paymentMethodDetails: details,
-                                                amount: context.payment?.amount,
-                                                order: nil)
     }
 
     /// Generate the payment details and invoke PaymentsComponentDelegate method.
     public func initiatePayment() {
         submit(data: paymentData)
     }
+
+    // MARK: - LoadingComponent
+
+    public func stopLoading() {
+        submitButtonItem.showsActivityIndicator = false
+        formViewController.view.isUserInteractionEnabled = true
+    }
+
+    // MARK: - Private
+
+    private var shouldStorePaymentMethod: Bool? {
+        configuration.showsStorePaymentMethodField ? storeDetailsItem.value : nil
+    }
+
+    private func didSelectSubmitButton() {
+        guard formViewController.validate() else { return }
+
+        startLoading()
+        initiatePayment()
+    }
+
+    private func startLoading() {
+        submitButtonItem.showsActivityIndicator = true
+        formViewController.view.isUserInteractionEnabled = false
+    }
 }
 
 @_spi(AdyenInternal)
 extension TwintComponent: PaymentInitiable {}
+
+@_spi(AdyenInternal)
+extension TwintComponent: PresentableInitiableComponent {
+
+    public var requiresPresentation: Bool {
+        configuration.showsStorePaymentMethodField
+    }
+}
+
+@_spi(AdyenInternal)
+extension TwintComponent: ViewControllerDelegate {}
```

Comment:
I think this component now also needs to be trackable if it's presented. Let's discuss

Reply:
Done!